scripts/loud.coffee

```diff
@@ -21,27 +21,27 @@ module.exports = (robot) ->
   robot.hear /^([A-Z"][A-Z0-9 .,'"()\?!&%$#@+-]+)$/, (res) ->
     # Pick a loud from the stored list and say it. Skip if there are no louds.
     old_loud = res.random(robot.brain.get('louds'))
+
     if old_loud?
       res.send old_loud
 
     # Save new loud in the list, but only if it is unique
     new_loud = res.match[1].trim()
     if new_loud not in robot.brain.get('louds')
-      robot.brain.get('louds').push(new_loud)
+      robot.brain.get('louds').push(new_loud) if new_loud not in robot.brain.get('louds_banned')
 
   robot.respond /loud (\w+)\s*(.*)?$/i, (res) ->
     action = res.match[1].trim()
     data = res.match[2]?.trim()
 
-    switch action 
+    switch action
       when 'delete'
-        index = robot.brain.get('louds').indexOf(data)
-        if index != -1
-          robot.brain.get('louds').splice(index, 1)
-          res.send "Loud deleted."
-        else
-          res.send "Couldn't find that loud."
-    
+        deleteLoud(data)
+
+      when 'ban'
+        robot.brain.get('louds_banned').push(data) if not data in robot.brain.get('louds_banned')
```

Comment:
Looks like `not in` is a better operator choice here. Sorry for the :crap: :spaghetti: